R/write_sessioninfo.R

```diff
@@ -0,0 +1,71 @@
+#' Write essential session info to a file or connection
+#'
+#' Writes essentials of
+#' \code{\link[sessioninfo:session_info]{sessioninfo::session_info()}}
+#' to a file or connection in order to
+#' maintain versionable documentation of a workflow and aid
+#' reproducibility.
+#'
+#' The following information is written:
+#' \itemize{
+#' \item{essential platform info: version, os, system, ctype}
+#' \item{essential info on loaded packages: package, loadedversion, date,
+#' source if different from CRAN.}
+#' }
+#'
+#' @param file A file path or a \code{\link[base]{connection}}.
+#'
+#' @importFrom sessioninfo
+#' platform_info
+#' package_info
+#' @importFrom stringr
+#' str_detect
+#' @importFrom utils
+#' write.table
+#' capture.output
+#' @importFrom rlang
+#' .data
+#' @importFrom dplyr
+#' mutate
+#' select
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' write_sessioninfo(file = "./sessioninfo.txt")
+#' }
+write_sessioninfo <- function(file) {
```

Comment:
The function writes both platform info (tab separated) and package info (space separated) to the same text file. Whether this is OK depends a bit on the workflow you have in mind. An alternative might be to return package info as a data.frame (or file) with platform info as attributes or as a separate data.frame (or file)?

Reply:
I just wanted a simple text file that holds this info in a specific format and that can be versioned (typically with git). The dataframe approach generally needs R to inspect dataframe contents and differences (but depending on how you store it, you could as well inspect the associated file directly: csv, tsv, ...). In this case I see the dataframe format as an unneeded complexity - do you want to load it as dataframe back into R?

It is true that the precise formatting of the file is rather arbitrary, alternatives may work just as well.

Reply:
I agree with your take on it.